lib/semian.rb

```diff
@@ -156,6 +157,14 @@ def register(name, **options)
   end
 
   def retrieve_or_register(name, **args)
+    # If consumer who retrieved / registered by a Semian::Adapter, keep track
+    # of who the consumer was so that we can clear the resource reference if needed.
+    if consumer = args.delete(:consumer)
+      if consumer.class.included_modules.include?(Semian::Adapter)
```

Comment:
@dalehamel in your example you invoke it on the class, whereas in the code we're commenting you call `.class` on the instance.

`instance.class.include?(Module)` vs `instance.is_a?(Module)`